.teamcity/_self/projects/WebApp.kt

```diff
@@ -370,6 +370,18 @@ object RunAllUnitTests : BuildType({
 				yarn test-packages --maxWorkers=${'$'}JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-teamcity --silent
 			"""
 		}
+		bashNodeScript {
+			name = "Ensure prepack works correctly"
+			executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
+			scriptContent = """
+				unset NODE_ENV
+				unset CALYPSO_ENV
+
+				# Tests that yarn run prepack works. This tests the npm publish flow slightly.
+				yarn workspaces foreach run prepare
```

Comment:
Should we make sure that all workspaces actually have the `prepare` and `prepack` scripts if we're going to run them on all of the workspaces?

Reply:
In my testing, it skips a workspace if it doesn't have the script:

```sh
~/source/wp-calypso trunk ⇣ ≡
$ yarn --verbose workspaces foreach run prepare
➤ YN0000: [@automattic/calypso-color-schemes]: Process started
➤ YN0000: [@automattic/calypso-color-schemes]: Process exited (exit code 0), completed in 1s 169ms

➤ YN0000: [@automattic/calypso-config]: Process started
➤ YN0000: [@automattic/calypso-config]: Process exited (exit code 0), completed in 1s 536ms

➤ YN0000: [@automattic/create-calypso-config]: Process started
➤ YN0000: [@automattic/create-calypso-config]: Process exited (exit code 0), completed in 1s 478ms

➤ YN0000: [@automattic/languages]: Process started
➤ YN0000: [@automattic/languages]: Process exited (exit code 0), completed in 1s 524ms
➤ YN0000: Done in 5s 717ms
```

And if you search for "prepare" in all our package.json files, only these four workspaces have the script. And if you run `yarn --verbose workspaces foreach run prepack`, a different set of workspaces run.

So i think this should work ok. Plus, it doesn't make sense for things not published to npm to have those steps necessarily

Reply:
I agree, but I did notice that the "unit tests" check fails - any idea what might be causing that?